pkg/daemon/ceph/client/pool.go

```diff
@@ -303,29 +306,38 @@ func CreateECPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, poo
 	return nil
 }
 
-func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, poolName string, pool cephv1.PoolSpec, pgCount, appName string, cluster cephv1.ClusterSpec) error {
-	// Create a CRUSH rule for stretched clusters
-	if pool.Replicated.ReplicasPerFailureDomain != 0 {
-		err := createStretchedReplicationCrushRule(context, clusterInfo, poolName, pool, cluster)
-		if err != nil {
-			return errors.Wrap(err, "failed to create stretched replicated crush rule")
-		}
+func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, poolName string, pool cephv1.PoolSpec, pgCount, appName string) error {
+	// The crush rule name is the same as the pool unless we have a stretch cluster.
+	crushRuleName := poolName
+	if clusterSpec.IsStretchCluster() {
+		// Stretch clusters usually all use the same default crush rule
+		crushRuleName = defaultStretchCrushRuleName
 	} else {
-		// create a crush rule for a replicated pool, if a failure domain is specified
-		if err := createReplicationCrushRule(context, clusterInfo, poolName, pool, cluster); err != nil {
-			return errors.Wrap(err, "failed to create replicated crush rule")
+		// Create a CRUSH rule for stretched clusters
+		if pool.Replicated.ReplicasPerFailureDomain != 0 {
```

Comment:
Don't we already check that earlier?

Reply:
I'll clarify this is comment when renaming `createStretchedReplicationCrushRule()`